book-service/src/test/java/com/kadet/best/book/BooksCreateRestTest.java

```diff
@@ -0,0 +1,199 @@
+        Book.class
+    );
+    assertThat(createdBook.getId()).isNull();
+    assertThat(createdBook.getName()).isEqualTo("Sims 3");
+    assertThat(createdBook.getDescription()).isEqualTo("Game for girls");
+
+    // 2. Delete book
+    deleteBook(createBookResult.andReturn().getResponse().getHeader(HttpHeaders.LOCATION));
+  }
+
+  @Test
+  public void should_create_book_with_only_name() throws Exception {
+    // 1. Create book
+    // given
+    val bookToCreate = new Book()
+        .setName("Sims 3");
+
+    // when
+    final ResultActions createBookResult = mockMvc.perform(post("/books")
+        .content(objectMapper.writeValueAsString(bookToCreate)));
+
+    // then
+    createBookResult.andExpect(status().isCreated())
+        .andExpect(header().string(HttpHeaders.LOCATION, containsString("books/")));
+
+    val createdBook = objectMapper.readValue(
+        createBookResult.andReturn().getResponse().getContentAsByteArray(),
+        Book.class
+    );
+    assertThat(createdBook.getId()).isNull();
+    assertThat(createdBook.getName()).isEqualTo("Sims 3");
+    assertThat(createdBook.getDescription()).isNull();
+
+    // 2. Delete book
+    deleteBook(createBookResult.andReturn().getResponse().getHeader(HttpHeaders.LOCATION));
+  }
+
+  @Test
+  public void should_create_book_with_only_description() throws Exception {
+    // 1. Create book
+    // given
+    val bookToCreate = new Book()
+        .setDescription("My description");
+
+    // when
+    final ResultActions createBookResult = mockMvc.perform(post("/books")
+        .content(objectMapper.writeValueAsString(bookToCreate)));
+
+    // then
+    createBookResult.andExpect(status().isCreated())
+        .andExpect(header().string(HttpHeaders.LOCATION, containsString("books/")));
+
+    val createdBook = objectMapper.readValue(
+        createBookResult.andReturn().getResponse().getContentAsByteArray(),
+        Book.class
+    );
+    assertThat(createdBook.getId()).isNull();
+    assertThat(createdBook.getName()).isNull();
+    assertThat(createdBook.getDescription()).isEqualTo("My description");
+
+    // 2. Delete book
+    deleteBook(createBookResult.andReturn().getResponse().getHeader(HttpHeaders.LOCATION));
+  }
+
+  @Test
+  public void should_create_book_with_no_values_when_id_is_defined() throws Exception {
+    // 1. Create book
+    // given
+    val bookToCreate = new Book()
+        .setId("534687435");
+
+    // when
+    final ResultActions createBookResult = mockMvc.perform(post("/books")
+        .content(objectMapper.writeValueAsString(bookToCreate)));
+
+    // then
+    createBookResult.andExpect(status().isCreated())
+        .andExpect(header().string(HttpHeaders.LOCATION, containsString("books/")));
+
+    val createdBook = objectMapper.readValue(
+        createBookResult.andReturn().getResponse().getContentAsByteArray(),
+        Book.class
+    );
+    assertThat(createdBook.getId()).isNull();
```

Comment:
http://stackoverflow.com/questions/24936636/while-using-spring-data-rest-after-migrating-an-app-to-spring-boot-i-have-obser